src/registry/domain/require-wrapper.js

```diff
@@ -1,32 +1,44 @@
 'use strict';
 
+const coreModules = require('builtin-modules');
 const path = require('path');
 const _ = require('lodash');
 const requirePackageName = require('require-package-name');
 
 const strings = require('../../resources');
 
-module.exports = function(injectedDependencies){
-  return function(requirePath){
-    const moduleName = requirePackageName(requirePath);
+const getError = (moduleName) => ({
+  code: strings.errors.registry.DEPENDENCY_NOT_FOUND_CODE,
+  missing: [moduleName]
+});
 
-    if(!_.includes(injectedDependencies, moduleName)){
-      throw {
-        code: strings.errors.registry.DEPENDENCY_NOT_FOUND_CODE,
-        missing: [moduleName]
-      };
-    }
+const requireCoreDependency = (requirePath) => {
+  if(_.includes(coreModules, requirePath)){
+    return require(requirePath);
+  }
+  throw new Error();
+};
+
+const requireDependency = (requirePath) => {
+  const nodeModulesPath = path.resolve('.', 'node_modules');
+  const modulePath = path.resolve(nodeModulesPath, requirePath);
+  return require(modulePath);
```

Comment:
I'll perhaps move the try catch logic inside each requiring function `requireCoreDependency` and `requireCoreDependency` .